substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -949,20 +979,54 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 		} else {
 			self.ext.get_storage(&key)
 		};
+
 		if let Some(value) = outcome {
 			self.adjust_gas(charged, costs(value.len() as u32));
-			self.write_sandbox_output(
-				memory,
-				out_ptr,
-				out_len_ptr,
-				&value,
-				false,
-				already_charged,
-			)?;
-			Ok(ReturnErrorCode::Success)
+
+			match read_mode {
+				StorageReadMode::ZeroIfMissing => {
+					let mut fixed_output = [0u8; 32];
+					let len = core::cmp::min(value.len(), 32);
+					fixed_output[..len].copy_from_slice(&value[..len]);
+
+					self.write_fixed_sandbox_output(
+						memory,
+						out_ptr,
+						&fixed_output,
+						false,
+						already_charged,
+					)?;
+					Ok(ReturnErrorCode::Success)
+				},
+				StorageReadMode::ErrorIfMissing => {
+					// preserve original behavior
```

Comment:
This comment doesn't make sense here. From the point of looking at the code nobody will know what the old behaviour was. Better document this on the enum.